src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -1119,17 +1120,21 @@ public Solution WithProjectAnalyzerReferences(ProjectId projectId, IEnumerable<A
 
         private Solution AddDocument(DocumentState state)
```

Comment:
Is there no way to pull this out to the larger project load/unload, rather than doing this a bunch of times?